plugins/user-authenticators/ldap/src/main/java/org/apache/cloudstack/api/command/LdapListUsersCmd.java

```diff
@@ -201,33 +199,10 @@ public void execute() throws ServerApiException {
         if(s_logger.isTraceEnabled()) {
             s_logger.trace(String.format("applying filter: %s or %s.", this.getListTypeString(), this.getUserFilter()));
         }
-        Method filterMethod = getFilterMethod();
-        List<LdapUserResponse> responseList = ldapResponses;
-        if (filterMethod != null) {
-            if(s_logger.isTraceEnabled()) {
-                s_logger.trace("applying filter: " + filterMethod.getName());
-            }
-            try {
-                responseList = (List<LdapUserResponse>)filterMethod.invoke(this, ldapResponses);
-            } catch (IllegalAccessException | InvocationTargetException e) {
-                throw new CloudRuntimeException("we're damned, the earth is screwed. we will now return to our maker",e);
-            }
-        }
+        List<LdapUserResponse> responseList = getUserFilter().filter(this,ldapResponses);
```

Comment:
Looks better but I would prefer having an if-then-else block here instead of the abstract method implemented on the enum

Reply:
This refactoring LGTM, @nvazquez @DaanHoogland.

Reply:
tnx @rhtyd. @nvazquez I am not changing this. Having functionality specific to an enum value be implemented by that enum value is a more neat solution than having a switch in calling code. we'll have a beer about it, next conf.